test/test_api/test_project_locks.py

```diff
@@ -0,0 +1,197 @@
+import json
+from unittest.mock import patch, call
+
+from nose.tools import assert_equal
+
+from unittest.mock import patch
+
+from pybossa.model.project import Project
+from pybossa.api.project_locks import ProjectLocksAPI
+from test import db, with_context
+from test.factories import (ProjectFactory, UserFactory)
+from test.test_api import TestAPI
+
+
+class TestProjectAPI(TestAPI):
+
+    def setUp(self):
+        super(TestProjectAPI, self).setUp()
+        db.session.query(Project).delete()
+
+    def setupProjects(self):
+        project = ProjectFactory.create(
+            updated='2015-01-01T14:37:30.642119',
+            short_name='test-app1',
+            info={
+                'total': 150,
+                'task_presenter': 'foo',
+                'data_classification': dict(input_data="L4 - public", output_data="L4 - public"),
+                'product' : 'test_product',
+                'subproduct': 'test_subproduct1'
+        })
+
+        projects = ProjectFactory.create_batch(5,
+            info={
+                'total': 150,
+                'task_presenter': 'foo',
+                'data_classification': dict(input_data="L4 - public", output_data="L4 - public"),
+                'product' : 'test_product',
+                'subproduct': 'test_subproduct2'
+        })
+        return project
+
+    def _create_dict_from_model(self, item):
+        # Dummy method for testing
+        return {}
+
+    def _sign_item(self, item):
+        # Dummy method for testing
+        pass
+
+    @with_context
+    def test_project_locks_user_not_logged_in(self):
+        """ Test should return 401 if the user is not logged in"""
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks/' + project_id)
+        err = json.loads(res.data)
+        assert res.status_code == 401, err
+        assert err['status'] == 'failed', err
+        assert err['target'] == 'project', err
+        assert err['exception_cls'] == 'Unauthorized', err
+        assert err['action'] == 'GET', err
+
+    @with_context
+    def test_project_locks_user_worker(self):
+        """ Test API should return 401 if user is worker"""
+        admin = UserFactory.create(admin=True)
+        worker = UserFactory.create(admin=False, subadmin=False)
+
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks?id=' + project_id + '&api_key=' + worker.api_key + '&all=1')
+        err = json.loads(res.data)
+        assert res.status_code == 401, err
+        assert err['status'] == 'failed', err
+        assert err['target'] == 'project', err
+        assert err['exception_cls'] == 'Unauthorized', err
+        assert err['action'] == 'GET', err
+
+    @with_context
+    def test_project_locks_user_subadmin(self):
+        """ Test API should work if user is subadmin"""
+        admin = UserFactory.create(admin=True)
```

Comment:
like above coment, `admin` is also not used

Reply:
Fixed in https://github.com/bloomberg/pybossa/pull/933/commits/86953d171ad5e6bd1a93c8e144d76862270f1484.